bonsai-core/src/main/scala/com/stripe/bonsai/FullBinaryTree.scala

```diff
@@ -92,6 +96,22 @@ class FullBinaryTree[A, B](
   final def nonEmpty: Boolean = bitset(0)
   final def isEmpty: Boolean = !bitset(0)
 
+  override def equals(that: Any): Boolean = that match {
+    case (that: FullBinaryTree[_, _]) =>
+      this.bitset == that.bitset &&
+      this.isLeaf == that.isLeaf &&
+      this.branchLabels == that.branchLabels &&
+      this.leafLabels == that.leafLabels
+
+    case _ =>
+      false
+  }
+
+  override def hashCode: Int = {
```

Comment:
Can we get some tests that equals and hashCode agree? I think this applies to `FullBinaryTree`, `IndexedBitSet`, and `Vec`.